### ISSUE TITLE
Rule idea: no useless template literals

### ISSUE THREAD

**Issue by danvk** [eslint/eslint#6614]
**When does this rule warn? Please describe and show example code:**

```javascript
// These are bad:
const foo = `${otherVar}`;
const bar = `${someFn(1, 2, 3)}`;

// They should be:
const foo = otherVar;
const bar = someFn(1, 2, 3);
```

**Is this rule preventing an error or is it stylistic?**

Stylistic.

**Why is this rule a candidate for inclusion instead of creating a custom rule?**

I noticed this pattern a bunch in a recent code review. It can easily happen if you're modifying existing code that uses template literals and don't notice that your literals have become trivial.

**Are you willing to create the rule yourself?**

Possibly. (I've implemented tslint rules but never eslint rules.)

**Comment by danvk** [comment]
Worth noting that one side effect of the "useless" template literals is that they convert whatever they contain to a string. So if `someFn(1, 2, 3)` returns a number in the example above, then the two `bar`s will have different types (string and number).

**Comment by platinumazure** [comment]
My personal take is that this might not merit inclusion as a core rule unless we can see evidence that this is a serious problem (or an important component of a popular style guide). I think this would be great as a custom rule or a plugin rule. But that's just my take-- we can see what the rest of the team thinks.

**Comment by alberto** [comment]
As you mentioned they are not equivalent, since the template ensures you get a string.

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
